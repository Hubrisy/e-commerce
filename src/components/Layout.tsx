import Footer from '@/components/footer';
import Header from '@/components/header';

export const Layout = ({ children }: { children: React.ReactNode }) => (
  <>
    <Header />
    <main>{children}</main>
    <Footer />
  </>
);

export default Layout;
