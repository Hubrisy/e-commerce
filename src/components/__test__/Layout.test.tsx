import { render } from '@testing-library/react';

import { Layout } from '../Layout';

jest.mock('@/components/header', () => ({
  __esModule: true,
  default: () => <div>Header Mock</div>,
}));

jest.mock('@/components/footer', () => ({
  __esModule: true,
  default: () => <div>Footer Mock</div>,
}));

describe('Layout', () => {
  test('renders correctly', () => {
    const children = <div>Test Children</div>;
    const { asFragment } = render(<Layout>{children}</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });
});
