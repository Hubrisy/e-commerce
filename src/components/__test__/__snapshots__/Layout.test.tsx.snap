// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<DocumentFragment>
  <div>
    Header Mock
  </div>
  <main>
    <div>
      Test Children
    </div>
  </main>
  <div>
    Footer Mock
  </div>
</DocumentFragment>
`;
